modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1054 @@
+  fcn.args$source                 <- source
+  fcn.args$collection             <- collection
+  fcn.args$siteid                 <- siteid_short
+  fcn.args$scale                  <- as.double(scale)
+  fcn.args$projection             <- projection
+  fcn.args$qc                     <- as.double(qc)
+  fcn.args$algorithm              <- algorithm
+  fcn.args$input_file             <- input_file
+  fcn.args$credfile               <- credfile
+  fcn.args$out_get_data           <- out_get_data
+  fcn.args$out_process_data       <- out_process_data
+  fcn.args$stage_get_data         <- stage_get_data
+  fcn.args$stage_process_data     <- stage_process_data
+  fcn.args$raw_merge              <- raw_merge
+  fcn.args$pro_merge              <- pro_merge
+  fcn.args$existing_raw_file_path <- existing_raw_file_path
+  fcn.args$existing_pro_file_path <- existing_pro_file_path
+  
+  
+  arg.string <- PEcAn.utils::listToArgString(fcn.args)
+  
+  cmdFcn <- paste0("RpTools$rp_control(", arg.string, ")")
+  PEcAn.logger::logger.debug(paste0("Remote module executing the following function:\n", cmdFcn))
+  
+  # call rp_control
+  output <- do.call(RpTools$rp_control, fcn.args)
+
+  # insert output data in the DB
+  db_out <-
+    remotedata_db_insert(
+      output                = output,
+      remotefile_check_flag = remotefile_check_flag,
+      siteid                = siteid,
+      out_get_data          = out_get_data,
+      out_process_data      = out_process_data,
+      write_raw_start       = write_raw_start,
+      write_raw_end         = write_raw_end,
+      write_pro_start       = write_pro_start,
+      write_pro_end         = write_pro_end,
+      raw_check             = raw_check,
+      pro_check             = pro_check,
+      raw_mimetype          = raw_mimetype,
+      raw_formatname        = raw_formatname,
+      pro_mimetype          = pro_mimetype,
+      pro_formatname        = pro_formatname,
+      dbcon                 = dbcon
+    )
+  
+  
+  # return the ids and paths of the inserted data
+  if (!is.null(out_get_data)) {
+    settings$remotedata$raw_id   <- db_out[[1]]
+    settings$remotedata$raw_path <- db_out[[2]]
+  }
+  if (!is.null(out_process_data)) {
+    settings$remotedata$pro_id   <- db_out[[3]]
+    settings$remotedata$pro_path <- db_out[[4]]
+  }
+  
+  return (settings)
+}
+
+
+
+##' construct the raw file name
+##'
+##' @name construct_raw_filename
+##' @title construct_raw_filename
+##' @param collection collection or product requested from the source
+##' @param siteid shortform of siteid
+##' @param scale scale, NULL by default
+##' @param projection projection, NULL by default
+##' @param qc qc_parameter, NULL by default
+##' @return raw_file_name
+##' @examples
+##' \dontrun{
+##' raw_file_name <- construct_raw_filename(
+##'   collection="s2",
+##'   siteid=721,
+##'   scale=10.0
+##'   projection=NULL
+##'   qc=1.0)
+##' }
+##' @author Ayush Prasad
+construct_raw_filename <-
```

Comment:
could you modify this function such as the following lines so that we don't have `NA`s in the filenames, and also add the source to filename:

```
construct_raw_filename <-
  function(source,
           collection,
           siteid,
           scale = NULL,
           projection = NULL,
           qc = NULL) {
    # skip if a parameter is not applicable and is NULL
    if (is.null(scale)) {
      scale_str <- "_"
    } else{
      scale_str <- paste0("_", format(scale, nsmall = 1), "_")
    }
    if (is.null(projection)) {
      prj_str <- ""
    }else{
      prj_str <- paste0(projection, "_")
    }
    if (is.null(qc)) {
      qc_str <- ""
    } else{
      qc_str <- paste0(format(qc, nsmall = 1), "_")
    }
    raw_file_name <-
      paste0(toupper(source), "_", collection, scale_str, prj_str, qc_str, "site_", siteid)
```